examples/appservice_autojoin/src/main.rs

```diff
@@ -42,8 +42,10 @@ pub async fn handle_room_member(
 pub fn error_if_user_not_in_use(error: matrix_sdk_appservice::Error) -> Result<()> {
     match error {
         // If user is already in use that's OK.
-        matrix_sdk_appservice::Error::Matrix(matrix_sdk::Error::Http(HttpError::UiaaError(
-            FromHttpResponseError::Server(ServerError::Known(UiaaResponse::MatrixError(error))),
+        matrix_sdk_appservice::Error::Matrix(matrix_sdk::Error::Http(HttpError::Api(
+            FromHttpResponseError::Server(ServerError::Known(RumaApiError::Uiaa(
+                UiaaResponse::MatrixError(error),
+            ))),
```

Comment:
I wonder how this PR is an improvement while the nesting here got even deeper?!

Reply:
For me https://github.com/matrix-org/matrix-rust-sdk/pull/1097/files#diff-ea840e928dbe5a05ef8c4efcbe859fec31c5b97690d5e2879a96ebe31b00128eR39 looks like an actual improvement. :shrug:

Reply:
The first commit doesn't make code that digs into error variants any shorter, that is right. However, it makes it so you only have to check one variant (`Api`) instead of two (`Api`, `UiaaError`) when you want to check for server-side errors.

Reply:
I have now added a commit that makes use of `as_ruma_api_error` internally, and this became much shorter as a result.

Reply:
From the consumer ergonomic perspective this still seems far from nice to use. Now it does a method call in-between instead of a nested match, essentially being equally deep nested and not obvious to use. And more LOC than before.